### PR TITLE
add routes for players personal info and type player

### PR DESCRIPTION
**Pull request by fernandoamz**
Routes Api for players personal info and type player 

captura de pantalla 2018-11-12 a la s 23 52 51

Unit testing

captura de pantalla 2018-11-14 a la s 11 42 49

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# Note that this schema.rb definition is the authoritative source for your
```
Comment (archisss) [inline db/schema.rb]: you can't delete all this comment lines

File: app/controllers/api/players_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    if @player.destroy
+        render json: @player, status: :deleted
+    else
+        render json: @player, status: :error
+    end
+  end
+
+  private 
+  def player_params
+    params.permit(:user_id,:id_user_type,:country_residence,:state_residence,:city_residence,:nationality,:state_birth,:city_birth,:persona_description,:weight,:height,:preferred_position,:telephone,:achievements)
```
Comment (AlejandroHdzM) [inline app/controllers/api/players_controller.rb]: What do you mean with `:persona_description` also avoid using Spanish words

File: app/controllers/api/players_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class Api::PlayersController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  def index 
+    @players = Player.all
```
Comment (AlejandroHdzM) [inline app/controllers/api/players_controller.rb]: Why are you getting all players?

Comment (fernandoamz) [reply app/controllers/api/players_controller.rb]: this query, is for profile visor, he can view all players and i add cause is part of CRUD

File: app/controllers/api/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+class Api::SessionsController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def show 
+    current_user ? head(:ok) : head(:unauthorized)
+  end
+
+  def create 
+    user = User.where(email: params[:email]).first
```
Comment (diegosaul402) [inline app/controllers/api/sessions_controller.rb]: If you're looking for just one record, is better to use `user =  User.find_by(email: params[:email])`
check this doc for more info https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find

File: app/controllers/api/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      head(:unauthorized)
+    end
+  end
+
+  def destroy
+    current_user&.authentication_token = nil
+    if current_user.save
+      head(:ok)
+    else
+      head(:unauthorized)
```
Comment (diegosaul402) [inline app/controllers/api/sessions_controller.rb]: This error message is wrong, the problem is not the user is unauthorized, please read an follow this useful guide for error handling, your front end will thank you 
https://nordicapis.com/best-practices-api-error-handling/

File: spec/models/player_spec.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    it 'ensures player model save' do
+      player = Player.new(user_id: 1, id_user_type: 1, country_residence: 2, state_residence: 1, city_residence: 3, nationality: 'MEX', state_birth: 12, city_birth: 2,persona_description: 'bla bla', weight: 80, height: 70, preferred_position: 'delantero',telephone: '1234567', achievements: 'testing').save
+      expect(player).to eq(true)
+    end
+  end
+
+  context 'model is updating' do 
+    it 'ensures that model is updating elements' do
+      player = Player.find_by(id:8).update(telephone: '123345')
+      expect(player).to eq(true)
```
Comment (diegosaul402) [inline spec/models/player_spec.rb]: This test should check if the new phone number is actually recorded for that player.

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+  context 'User Model is working' do
+    it 'ensures that User model is saving users' do
+      user = User.new(email: 'email@email.com', password: 'testing', password_confirmation: 'testing').save
+      expect(user).to eq(true)
+    end
+
+    it 'ensures that User model is updating users' do
+      user = User.find_by(id:7).update(password: '12345678')
+      expect(user).to eq(true)
```
Comment (diegosaul402) [inline spec/models/user_spec.rb]: Same about this and all others tests about updating.

File: spec/requests/logins_spec.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Logins', type: :request do
+  headers = {
+    'ACCEPT' => 'application/json'
+  }
+
+  it 'POST Login session' do
+    post '/api/sessions', params: { email: 'testing@tester.com', password: '1234567' }, :headers => headers
```
Comment (diegosaul402) [inline spec/requests/logins_spec.rb]: Please update your hash syntax here.

File: spec/requests/players_spec.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    end
+  end 
+
+  describe 'POST create /players' do
+    it 'POST create personal info player' do
+      post '/api/players', params: {  user_id: 1, id_user_type: 1, country_residence: 2, 
+                                    state_residence: 1, city_residence: 3, nationality: 'MEX', 
+                                    state_birth: 12, city_birth: 2,personal_description: 'bla bla', 
+                                    weight: 80, height: 70, preferred_position: 'delantero', telephone: '1234567',
+                                    achievements: 'testing' }, :headers => headers
```
Comment (diegosaul402) [inline spec/requests/players_spec.rb]: same about hash syntax. Shouldn't be any arrow `=>` on the code

File: test/controllers/api/players_controller_test.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class Api::PlayersControllerTest < ActionDispatch::IntegrationTest
```
Comment (diegosaul402) [inline test/controllers/api/players_controller_test.rb]: If your tests files does nothing is better don't commit them yet.

Comment (AlejandroHdzM) [reply test/controllers/api/players_controller_test.rb]: Those test are autogenerated by rails, do you suggest to delete it?

Comment (8geonirt) [reply test/controllers/api/players_controller_test.rb]: No, don't add this file to the `.gitignore`, if you do that then you have to create a commit ti make the changes be reflected and then when you start using it you'll have to remove it from the gitignore again, creating another commit. Also... if this was autogenerated you can leave the file as it is, there's no problem on pushing it (cc @diegosaul402)

**Pull request closed without merging**